core/GlobalState.cc

```diff
@@ -1211,6 +1211,9 @@ void GlobalState::_error(unique_ptr<Error> error) const {
         loc.file().data(*this).minErrorLevel_ = min(loc.file().data(*this).minErrorLevel_, error->what.minLevel);
     }
 
+    if (shouldReportErrorOn(loc, error->what)) {
```

Comment:
I feel that this is still too early. The thing on the other end of error queue may decide to ignore the error.